src/coreclr/src/tools/Common/TypeSystem/Common/TypeSystemContext.cs

```diff
@@ -778,12 +778,33 @@ internal TypeFlags ComputeTypeFlags(TypeDesc type, TypeFlags flags, TypeFlags ma
                 flags |= TypeFlags.HasStaticConstructorComputed;
             }
 
+            // We are looking to compute IsIDynamicInterfaceCastable and we haven't yet assigned a value
+            if ((mask & TypeFlags.IsIDynamicInterfaceCastableComputed) == TypeFlags.IsIDynamicInterfaceCastableComputed)
+            {
+                TypeDesc typeDefinition = type.GetTypeDefinition();
```

Comment:
We should add a `IsValueType` check to match the VM behavior.

(Also, let's not forget to document that implementing IDynamicInterfaceCastable on a valuetype will be silently ignored.)